networkx/algorithms/bipartite/link_analysis.py

```diff
@@ -0,0 +1,277 @@
+import itertools
+
+import networkx as nx
+
+__all__ = ["birank"]
+
+
+@nx._dispatchable(edge_attrs="weight")
+def birank(
+    G,
+    nodes,
+    alpha=None,
+    beta=None,
+    top_personalization={},
+    bottom_personalization={},
+    max_iter=100,
+    tol=1.0e-6,
+    weight="weight",
+):
+    r"""Compute the BiRank score for nodes in a bipartite network.
+
+    Given the bipartite sets $U$ and $P$, the BiRank algorithm seeks to satisfy
+    the following recursive relationships between the scores of nodes $j \in P$
+    and $i \in U$:
+
+    .. math::
+
+       p_j = \alpha \sum_{i \in U} \frac{w_{ij}}{\sqrt{d_i}\sqrt{d_j}} u_i
+       + (1 - \alpha) p_j^0
+
+       u_i = \beta \sum_{j \in P} \frac{w_{ij}}{\sqrt{d_i}\sqrt{d_j}} p_j
+       + (1 - \beta) u_i^0
+
+    where
+
+    * $p_j$ and $u_i$ are the BiRank scores of nodes $j \in P$ and $i \in U$.
+    * $w_{ij}$ is the weight of the edge between nodes $i \in U$ and $j \in P$
+      (With a value of 0 if no edge exists).
+    * $d_i$ and $d_j$ are the weighted degrees of nodes $i \in U$ and $j \in P$,
+      respectively.
+    * $p_j^0$ and $u_i^0$ are personalization values that can encode a priori
+      weights for the nodes $j \in P$ and $i \in U$, respectively. Akin to the
+      personalization vector used by PageRank.
+    * $\alpha$ and $\beta$ are damping hyperparameters applying to nodes in $P$
+      and $U$ respectively. They can take values in the interval $[0, 1]$, and
+      are analogous to those used by PageRank.
+
+    Below are two use cases for this algorithm.
+
+    1. Personalized Recommendation System
+        Given a bipartite graph representing users and items, BiRank can be used
+        as a collaborative filtering algorithm to recommend items to users.
+        Previous ratings are encoded as edge weights, and the specific ratings
+        of an individual user on a set of items is used as the personalization
+        vector over items. See the example below for an implementation of this
+        on a toy dataset provided in [1]_.
+
+    2. Popularity Prediction
+        Given a bipartite graph representing user interactions with items, e.g.
+        commits to a GitHub repository, BiRank can be used to predict the
+        popularity of a given item. Edge weights should encode the strength of
+        the interaction signal. This could be a raw count, or weighted by a time
+        decay function like that specified in Eq. (15) of [1]_. The
+        personalization vectors can be used to encode existing popularity
+        signals, for example, the monthly download count of a repository's
+        package.
+
+    Parameters
+    ----------
+    G : graph
+       A bipartite network
+
+    nodes : list or container
```

Comment:
```suggestion
    nodes : iterable of nodes
```

Reply:
Yeah - this change makes a lot of sense. Primary reason it was this in the first place is because the centrality algorithms I used as a template use that description.

A brief survey of the docs suggests that there are a fair few different argument descriptions for `nodes` across the various functions of the `bipartite` module. It may be valuable to standardize these? I'll open a discussion for it later - I'd be interested in taking it on.